### DISCUSSION TITLE
Write to a json file

### DISCUSSION THREAD

**Discussion by wassim** [vercel/next.js#17287] [Help]
I know that Vercel is serverless and has a readonly system but would it be somehow possible to write a small json file and read it afterwards with node?

I tried with:

```
export default async (req, res) => {
    ...
    fs.writeFileSync(path.resolve('./public/example.js'), JSON.stringify(data))
}

```
Doesn't seem to work...

**Comment by lfades** [comment]
Writing to local files or folders (like `public`) will only work at build time. If you need to persist something between functions it's better to use an external DB like Redis or similar, over using the file system.

**Comment by zaynelovecraft** [comment]
cant you just use Incremental Static Regeneration

reference 

https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration
